woocommerce/woocommerce
# Replace deprecated `page` methods

### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/woocommerce/issues/37832.

This PR updates all deprecated `page` methods like `page.click()` with more standard commands like `page.locator().click()`. It also includes minor changes like:

- Added missing `await` keywords in a few areas (this seems to be one cause of flakiness)
- Replaced `page.waitForLoadState` commands with web assertions in a few areas, as suggested in the Playwright docs. Replacing _all_ of them will be done on a separate task.
- Fixed several inaccurate assertions, and removed a few unnecessary ones I found along the way
- Removed the `create-variable-spec` to avoid conflicting with https://github.com/woocommerce/woocommerce/pull/38335

However, this PR does not change the locator arguments used. For example, in the code
```js
adminPage.fill( 'input[name="log"]', admin.username )
```
the argument `input[name="log"]` was kept during the update like so:
```js
adminPage.locator( 'input[name="log"]' ).fill( admin.username )
```

This PR tries its best not to make big changes to the locator arguments, only when necessary. Changing to more user-facing locators will be done on a different task.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

#### Testing locally

1. Run the usual commands for setting up the local wp-env environment
1. `cd` into `plugins/woocommerce`
1. Run all E2E tests `pnmp test:e2e-pw`. Make sure that they pass.
2. Also feel free to search through the `e2e-pw` folder for any remaining deprecated methods, using VS Code or any IDE. The list of deprecated methods are found in https://playwright.dev/docs/api/class-locator#deprecated and https://playwright.dev/docs/api/class-page#deprecated. There should be none left.

#### Testing on CI
1. Rerun the E2E tests a few times on CI to make sure that no new flakiness was introduced.
